ext/php8/ddtrace.c

```diff
@@ -433,6 +438,7 @@ static PHP_RINIT_FUNCTION(ddtrace) {
     // Things that should only run on the first RINIT
     int expected_first_rinit = 1;
     if (atomic_compare_exchange_strong(&ddtrace_first_rinit, &expected_first_rinit, 0)) {
+        zai_config_first_time_rinit();
```

Comment:
It _runs_ only once, but let's say there are two requests coming in nearly simultaneously after a restart; will the second request then possibly run before the first_time_rinit here has completed? I suspect this needs some sort of loop (or lock or whatever) at the end of this if() branch to be race-condition free.

Reply:
I recommend `pthread_once`. I'm using it in the profiler.

Reply:
Nice, didn't know about that one, looks good to me from the docs :-)